src/common-libs/auth/index.ts

```diff
@@ -1,14 +1,72 @@
-const AuthLib = {
-  async getRolesForAParticipant(participantId: number) {
-    return participantId; //TODO: add functionality
-  },
-
-  async getParticipantsForATarget(target: {
-    targetId: number;
-    target: string;
-  }) {
-    return target; //TODO: add functionality
-  },
+import { Request } from '@hapi/hapi';
+import { BASIC_AUTH_USER } from '@unocha/hpc-api-core/src/auth';
+
+interface BasicAuth {
+  username: string | null;
+  password: string | null;
+}
+
+const parseBasic = (credentialsPart: string): BasicAuth => {
+  let pieces: (string | null)[];
+
+  const decoded = Buffer.from(credentialsPart, 'base64').toString('utf8');
+
+  if (!decoded) {
+    throw new Error('Invalid authorization header');
+  }
+
+  const index = decoded.indexOf(':');
+
+  if (index === -1) {
+    pieces = [decoded];
+  } else {
+    pieces = [decoded.slice(0, index), decoded.slice(index + 1)];
+  }
+
+  // Allows for usernameless authentication
+  if (!pieces[0]) {
+    pieces[0] = null;
+  }
+
+  // Allows for passwordless authentication
+  if (!pieces[1]) {
+    pieces[1] = null;
+  }
+
+  return {
+    username: pieces[0],
+    password: pieces[1],
+  };
 };
 
-export default AuthLib;
+export const getTokenFromRequest = (req: Request): string | null => {
+  if (!req.headers?.authorization) {
+    return null;
+  }
+
+  const pieces = req.headers.authorization.split(/\s+/);
+
+  if (!pieces || pieces.length !== 2) {
+    throw new Error('Bad HTTP authentication header format');
+  }
+
+  const schemePart = pieces[0];
+  const credentialsPart = pieces[1];
+
+  switch (schemePart.toLowerCase()) {
+    case 'basic': {
+      const credentials = parseBasic(credentialsPart);
+
+      if (credentials.username !== BASIC_AUTH_USER) {
+        throw new Error('Client Authentication is not supported in the v4 API');
```

Comment:
This will produce a 500 error right? Let's use a BadRequest (400) error for this instead, as it is a misuse of the API that we should expect to need to handle.